### ISSUE TITLE
Spring Cleaning 2022 :broom: 

### ISSUE THREAD

**Issue by marcogario** [pysmt/pysmt#717]
Let's spend a few weeks to clean-up the project and focus on some key issues!

I propose we break this down into two phase:

### 1. Prioritizing (until June 1st - 2 weeks)
In the next ~2 weeks, we should focus on understanding what needs to be done. To achieve this, I created a [Project](https://github.com/orgs/pysmt/projects/1/). 

The goal is to triage all existing issues and PRs, and annotate them with one of the priorities below:
* 🔥 High - The project survival is at risk if we do not address this issue
* 🌞 Normal - Users are experiencing friction / pain when using pySMT because of this issue
* 💭 Low - This would be nice to have but it is more of a vitamin (extend pySMT) than a pain-killer (solve a pain point)

I expect we will have <5 issues marked as High and a <30% Normal.
For issues tracked as Low, please mark whether you suggest closing it. My expectations is that we will close ~50% of the open issues after the spring cleaning. Most will probably be closed due to lack of capacity to addressed them. This is totally fine, given the limited resources, we really want to focus on what matters.

This process will be async, and it should be possible to nibble at it by triaging a few issues in pockets of time. You can use [this filter](https://github.com/pysmt/pysmt/issues?q=is%3Aopen+-project%3Apysmt%2F1+) to know which issues/PR have not been triaged yet.

### 2. Fix all the things! (until June 20th - 3 weeks)

Once the triaging is done, we will start burning down our backlog starting from the High priority stuff. This will also occur asynchronously.

We will try to schedule a sync time close to the 20th to celebrate achievements, work out the last gnarly bugs, and agree on next steps for large remaining tasks.

### How to participate

Currently only @mikand and myself have write access to the project, and thus will mostly be us doing the prioritization (Phase 1). Note that anyone can read the Project, so hopefully this provides a good view of the direction we are taking.
If there is an issue that is dear to your heart, feel free to post a comment on it providing a suggested prioritization and justification for it. 

Phase 2 is open to anyone interested in contributing. Please subscribe to this issue to see progress updates and be involved in the conversation.

**Comment by marcogario** [comment]
I did a first pass through all the issues and triaged them. 

There are [27 issues](https://github.com/orgs/pysmt/projects/1/views/1?groupedBy%5BcolumnId%5D=4736342&filterQuery=propose-closing%3A%22%F0%9F%94%A5+Yes%22
) that I think we should just close as there is either not enough information to proceed, they are too complex to tackle without dedicated time, or they are covered by more general "upgrade the solvers" tasks that we do prior to a release.

@mikand could you go through these issues and do the honor of closing them if you agree?

There are 8 High priority issues (mostly around bugfixes) and 20 Normal priority. [See filter](https://github.com/orgs/pysmt/projects/1/views/1?groupedBy%5BcolumnId%5D=4736342). 

I think this gives us enough to start working at least on the High priority ones.

@mikand I will probably only have time to start working on this next week, so feel free to add/remove issues from the High priority group at anytime.

NOTE: I did not triage the PRs, but I think we can do that later.

**Comment by mikand** [comment]
@marcogario I closed some of the issues you marked. The one I left open are debatable: there are some things I think would be sensible to keep at least in the development plan....

**Comment by mikand** [comment]
Closing this tracking issue: the 2022 cleanup ran its course and the issue has
had no activity since. Triage is being redone now against current master, so
this one no longer reflects the state of the backlog.
